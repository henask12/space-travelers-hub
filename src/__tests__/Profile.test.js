// Profile.test.js
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // For custom DOM matchers
import MyProfilePage from '../components/Views';

// Mock empty missions and rockets data to be used by useSelector
// Mocking useDispatch and useSelector
jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
  useSelector: (selectorFn) => selectorFn(
    {
      rockets: { rockets: [] },
      missions: { missions: [] },
    },
  ),
}));

it('renders correctly with no missions and rockets', () => {
  // Render the component
  const { container } = render(<MyProfilePage />);

  // Assert that the component renders correctly
  expect(container).toMatchSnapshot();
});
