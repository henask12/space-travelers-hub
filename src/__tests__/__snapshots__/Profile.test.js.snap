// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no missions and rockets 1`] = `
<div>
  <section
    class="p-4 m-8"
  >
    <div
      class="flex"
    >
      <div
        class="w-1/2 pr-4"
      >
        <div
          class="bg-white border border-gray-380 rounded-lg shadow-lg p-4 hover:bg-gray-200"
        >
          <h1
            class="text-2xl font-semibold mb-4"
          >
            My Missions
          </h1>
          <table
            class="w-full border"
          >
            <thead>
              <tr
                class="bg-gray-100"
              >
                <th
                  class="px-4 py-2"
                >
                  Mission Name
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="px-4 py-2 text-center"
                >
                  <i>
                    No booked mission yet
                  </i>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="w-1/2 pl-4"
      >
        <div
          class="bg-white border-gray-380 rounded-lg shadow-lg p-4 border"
        >
          <h1
            class="text-2xl font-semibold mb-4"
          >
            My Rockets
          </h1>
          <table
            class="w-full border"
          >
            <thead>
              <tr
                class="bg-gray-100"
              >
                <th
                  class="px-4 py-2"
                >
                  Rocket Name
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="px-4 py-2 text-center"
                >
                  <i>
                    No rockets reserved yet
                  </i>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </section>
</div>
`;
